packages/app/src/app/components/WorkspaceSetup/steps/Finalize.tsx

```diff
@@ -1,13 +1,486 @@
+          </Text>
+        </Stack>
+
+        <Text color="#CCCCCC" size={4}>
+          {totalCredits} VM credits/month
+        </Text>
+
+        <Text color="#CCCCCC" size={4}>
+          Additional VM credits are available on-demand for $0.018/credit.
+          <br />
+          Spending limit: ${spendingLimit}
+        </Text>
+
+        {isAnnual ? (
+          <Stack
+            direction="vertical"
+            gap={4}
+            css={{ paddingBottom: 24 * 2, paddingTop: 24 * 2 }}
+          >
+            <Text color="#fff" size={4}>
+              Billing details:
+            </Text>
+
+            <Stack gap={2} direction="vertical">
+              <Text as="label" color="#e5e5e5" htmlFor="country">
+                Country *
+              </Text>
+
+              <Select
+                css={{ height: 40 }}
+                id="country"
+                value={country}
+                onChange={e => setCountry(e.target.value)}
+              >
+                <option value="">Select your country or region</option>
+                {COUNTRIES.map(item => (
+                  <option key={item} value={item}>
+                    {item}
+                  </option>
+                ))}
+              </Select>
+            </Stack>
+
+            <InputText
+              label="Postal code *"
+              placeholder="99999"
+              id="zipCode"
+              value={zipCode}
+              name="zipCode"
+              onChange={e => setZipCode(e.target.value)}
+            />
+          </Stack>
+        ) : (
+          <Stack />
+        )}
+
+        <Button
+          disabled={isAnnual && disabled}
+          autoWidth
+          size="large"
+          type="submit"
+          onClick={handleSubmit}
+        >
+          {isAnnual ? 'Send request' : 'Proceed to checkout'}
+        </Button>
+
+        <Stack gap={1} css={{ color: '#A8BFFA' }}>
+          <Icon name="circleBang" />
+          <Text size={3}>
+            {isAnnual
+              ? 'We will review your request in 24 hours and get back to you.'
+              : 'Process immediately.'}
+          </Text>
+        </Stack>
+      </Stack>
+    </AnimatedStep>
+  );
+};
 
 export const Finalize: React.FC<StepProps> = props => {
+  const { checkout } = useAppState();
   const { isPro } = useWorkspaceSubscription();
+
+  // This internal state is used to avoid unnecessary rerenders
+  // and show the wrong step while the user is recurring plan
+  const [_stepSelectedPlan, setStepSelectedPlan] = React.useState<PlanType>(
+    checkout.selectedPlan
+  );
+
+  if (_stepSelectedPlan === 'flex-annual') {
+    return (
+      <AnnualForm
+        refreshParentSelectedPlan={() =>
+          setStepSelectedPlan(checkout.selectedPlan)
+        }
+        {...props}
+      />
+    );
+  }
+
   if (isPro) {
     return <ChangePlan {...props} />;
   }
+
   return <Payment {...props} />;
 };
+
+const COUNTRIES = [
```

Comment:
just out of curiosity, where is this taken from? my geography geekiness noticed the list contains Palestine, but not Taiwan or Kosovo, which are both partially recognized sovereign states :D